File: Backend_Gateway/routes/logs.py
from fastapi import APIRouter
import requests
from datetime import datetime

router = APIRouter()

LOG_SERVICE = "http://localhost:5000"

@router.get("/")
def get_logs():

    response = requests.get(f"{LOG_SERVICE}/logs")

    requests.post(
        "http://localhost:5000/logs",
        json={
            "action": "GET_LOGS",
            "username": "bank_admin",
            "status": "SUCCESS",
            "timestamp": datetime.utcnow().isoformat()
        }
    )

    return response.json()

File: log-service/routes/logs.js
const express = require("express");
const router = express.Router();
const Log = require("../models/Log");

router.get("/", async (req, res) => {
    try {
        const logs = await Log.find();
        res.json(logs);
    } catch (err) {
        res.status(500).json({ message: err.message });
    }
});

router.post("/", async (req, res) => {
    try {
        const log = new Log({
            action: req.body.action,
            username: req.body.username,
            status: req.body.status,
            timestamp: new Date()
        });

        const savedLog = await log.save();

        res.status(201).json(savedLog);

    } catch (err) {
        res.status(500).json({
            message: err.message
        });
    }
});

module.exports = router;